presto-hive/src/test/java/com/facebook/presto/hive/TestCteExecution.java

```diff
@@ -0,0 +1,560 @@
+/*
```

Comment:
can you also add other types of queries like INSERT and create table, e.g.:
```
insert into r(a) with cte as (select 1) select * from cte;
 create table t as (with cte as (select 1 as a) select * from cte);
```

Reply:
also create view:
```
create view v as with cte as (select 1 as a) select * from cte;
```
What is the semantics of persisted CTEs in views? We should avoid persisting them during view definition

Reply:
Makes sense, I will add concrete test cases for insert, ctas and views. Will check for views, I think we only save view definition so mostly should be fine (will confirm).

Reply:
Confirmed that only definiations are stored with create views and added tests for all cases